### PR TITLE
Update Vendored Deps

### PR DESCRIPTION
**Pull request by jtlisi**
The vendored dependencies on this repo are very far behind. This aligns the vendored dependencies with Cortex master.

I updated the `rulefmt` used to be the package maintained by the `pkg/ruler/legacy_rulefmt` package in Cortex.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,25 +5,28 @@ go 1.12
...
-	google.golang.org/api v0.8.0
+	google.golang.org/api v0.14.0
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
-	gopkg.in/yaml.v2 v2.2.2
+	gopkg.in/yaml.v2 v2.2.8
 	gopkg.in/yaml.v3 v3.0.0-20200506231410-2ff61e1afc86
-	sigs.k8s.io/yaml v1.1.0
 )
+
+replace github.com/Azure/azure-sdk-for-go => github.com/Azure/azure-sdk-for-go v36.2.0+incompatible
```
Comment (gotjosh) [inline go.mod:30]: Why are these here?

Comment (gotjosh) [reply go.mod:30]: Let's add a comment on the why before merging if it's for a special reason

Comment (jtlisi) [reply go.mod:30]: These are overrides that are ported over from cortex due to some dependency issues there.
